diff --git a/sherpa-onnx/csrc/session.cc b/sherpa-onnx/csrc/session.cc
index 8206bb99..f0f6f7a1 100755
--- a/sherpa-onnx/csrc/session.cc
+++ b/sherpa-onnx/csrc/session.cc
@@ -368,15 +368,14 @@ Ort::SessionOptions GetSessionOptionsImpl(
       if (status) {
         const auto &api = Ort::GetApi();
         const char *msg = api.GetErrorMessage(status);
-        SHERPA_ONNX_LOGE(
-            "Failed to enable NNAPI: %s. Available providers: %s. Fallback to "
-            "cpu",
-            msg, os.str().c_str());
+        std::string error_msg = std::string("Failed to enable NNAPI: ") +
+                                msg + ". Available providers: " + os.str();
+        SHERPA_ONNX_LOGE("%s", error_msg.c_str());
         api.ReleaseStatus(status);
+        throw std::runtime_error(error_msg);
       } else {
         SHERPA_ONNX_LOGE("Use nnapi");
       }
